core/src/main/java/com/linecorp/armeria/internal/client/DefaultDnsNameResolver.java

```diff
@@ -103,10 +118,39 @@ public void operationComplete(Future<List<DnsRecord>> future) throws Exception {
             }
         };
 
-        questions.forEach(q -> delegate.resolveAll(q).addListener(listener));
+        final Builder<Promise<List<DnsRecord>>> promises =
+                ImmutableList.builderWithExpectedSize(questions.size());
+        questions.forEach(q -> {
+            final Promise<List<DnsRecord>> promise = executor().newPromise();
```

Comment:
I think we should either use `eventLoop` here or `executor()` below when scheduling for consistency.

Reply:
Oops. I removed `executor()` completely. Thanks!